source/contributing/implementation/debugging/git.rst

```diff
@@ -0,0 +1,34 @@
+===========
+Git Commits
+===========
+
+The first thing in debugging is determining which state your local repo needs to be at. You can use the ``stable-x`` 
+branch to learn the system and/or subsystems. However, researching a problem and working on a bug requires you to 
+revert the repo to a certain state in the past. For instance, you decide to debug an issue which specifies Forge 
+version: 2705 and SpongeForge version: 3371. You need to revert your repo to the point in time when the Sponge jar 
+file was created for SpongeForge version 3371.
```

Comment:
what's the usecase for this? if it can't be reproduced on the latest commit for a branch the issue is solved

Reply:
Oh, OK - I am used to re-creating the environment and reproducing the error in order to figure out what went wrong. But, you make a good point. I will remove this.

Reply:
What this may be reorganized as is "If you find a bug that is taking more time to resolve and your branch falls behind from the target, you can `git rebase` etc." sort of article. I figure that would be helpful to explain to contributing members when their bug fixes are large enough that they can't be merged without further testing.